aerogear/data-sync-ui
# Correct message when all datasources filtered out

https://issues.jboss.org/browse/AEROGEAR-7791
## Motivation
Better UX.

## What
Searching the data sources - when searching for an item and it's not found show a `No item found on search` type message.

## Why
As of writing this ticket it shows the blank slate message which could be confusing for the user.

## Verification Steps
1. Display page with datasources, when there is none, it should show `No Data Sources defined` page
2. Add some datasources
3. It should show the datasources in the list
4. Try to filter it by name so the filter no longer matches any of them
5. It should show page `No item found while filtering the data sources` and button `Clear the filter`
6. When pressing `Clear the filter` filter is cleared and all datasources are displayed

## Checklist:


## Progress

- [x] Displaying different messages
- [x] Possibility to clear filter quickly

## Additional Notes

PS.: Add images and/or .gifs to illustrate what was changed if this pull request modify the appearance/output of something presented to the users. 
<img src="https://user-images.githubusercontent.com/2912228/44271377-412bea00-a23a-11e8-9dc7-85abe20acb80.gif" width="800"/>
